Test/ServerTelemetryChannel.Test/Shared.Tests/SamplingTelemetryProcessorTest.cs

```diff
@@ -64,79 +65,188 @@ public void TelemetryItemHasSamplingPercentageSet()
         [TestMethod]
         public void DependencyTelemetryIsSubjectToSampling()
         {
-            TelemetryTypeSupportsSampling((telemetryProcessors) => telemetryProcessors.Process(new DependencyTelemetry()));
+            TelemetryTypeSupportsSampling(telemetryProcessors => telemetryProcessors.Process(new DependencyTelemetry()));
         }
         
         [TestMethod]
         public void EventTelemetryIsSubjectToSampling()
         {
-            TelemetryTypeSupportsSampling((telemetryProcessors) => telemetryProcessors.Process(new EventTelemetry("event")));
+            TelemetryTypeSupportsSampling(telemetryProcessors => telemetryProcessors.Process(new EventTelemetry("event")));
         }
         
         [TestMethod]
         public void ExceptionTelemetryIsSubjectToSampling()
         {
-            TelemetryTypeSupportsSampling((telemetryProcessors) => telemetryProcessors.Process(new ExceptionTelemetry(new Exception("exception"))));
+            TelemetryTypeSupportsSampling(telemetryProcessors => telemetryProcessors.Process(new ExceptionTelemetry(new Exception("exception"))));
         }
         
         [TestMethod]
         public void MetricTelemetryIsNotSubjectToSampling()
         {
-            TelemetryTypeDoesNotSupportSampling((telemetryProcessors) => telemetryProcessors.Process(new MetricTelemetry("metric", 1.0)));
+            TelemetryTypeDoesNotSupportSampling(telemetryProcessors =>
+            {
+                telemetryProcessors.Process(new MetricTelemetry("metric", 1.0));
+                return 1;
+            });
         }
         
         [TestMethod]
         public void PageViewTelemetryIsSubjectToSampling()
         {
-            TelemetryTypeSupportsSampling((telemetryProcessors) => telemetryProcessors.Process(new PageViewTelemetry("page")));
+            TelemetryTypeSupportsSampling(telemetryProcessors => telemetryProcessors.Process(new PageViewTelemetry("page")));
         }
         
         [TestMethod]
         public void PerformanceCounterTelemetryIsNotSubjectToSampling()
         {
             TelemetryTypeDoesNotSupportSampling(
-                (telemetryProcessors) => telemetryProcessors.Process(new PerformanceCounterTelemetry("category", "counter", "instance", 1.0)));
+                telemetryProcessors =>
+                {
+                    telemetryProcessors.Process(new PerformanceCounterTelemetry("category", "counter", "instance", 1.0));
+                    return 1;
```

Comment:
I did not want to change helper methods much. 1 means that 1 item was processed. In assert before there was a hardcoded expected number, now I use this return value and in some other tests call "Process" N times and return N.

---
In reply to: [67446947](https://github.com/Microsoft/ApplicationInsights-dotnet/pull/243#discussion_r67446947) [](ancestors = 67446947)